functions/helloworld/test/sample.integration.pubsub.test.js

```diff
@@ -14,38 +14,58 @@
  */
 
 // [START functions_pubsub_integration_test]
-const childProcess = require('child_process');
 const assert = require('assert');
+const execPromise = require('child-process-promise').exec;
+const delay = require('delay');
+const path = require('path');
+const supertest = require('supertest');
 const uuid = require('uuid');
 
-it('helloPubSub: should print a name', done => {
-  const startTime = new Date(Date.now()).toISOString();
+const request = supertest(process.env.BASE_URL || 'http://localhost:8080');
+const cwd = path.join(__dirname, '..');
+
+it('helloPubSub: should print a name', async () => {
   const name = uuid.v4();
 
-  // Mock Pub/Sub call, as the emulator doesn't listen to Pub/Sub topics
   const encodedName = Buffer.from(name).toString('base64');
-  const data = JSON.stringify({data: encodedName});
-  childProcess.execSync(`functions call helloPubSub --data ${data}`);
-
-  // Check the emulator's logs
-  const logs = childProcess
-    .execSync(`functions logs read helloPubSub --start-time ${startTime}`)
-    .toString();
-  assert.strictEqual(logs.includes(`Hello, ${name}!`), true);
-  done();
-});
-
-it('helloPubSub: should print hello world', done => {
-  const startTime = new Date(Date.now()).toISOString();
-
-  // Mock Pub/Sub call, as the emulator doesn't listen to Pub/Sub topics
-  childProcess.execSync('functions call helloPubSub --data {}');
-
-  // Check the emulator's logs
-  const logs = childProcess
-    .execSync(`functions logs read helloPubSub --start-time ${startTime}`)
-    .toString();
-  assert.strictEqual(logs.includes('Hello, World!'), true);
-  done();
-});
+  const pubsubMessage = {data: {data: encodedName}};
+
+  const proc = execPromise(
```

Comment:
`proc` is only used once. Remove it.

`const {stdout} = await execPromise(`

Reply:
Yes, that's fair. Let's comment to prevent linters or developers from inlining it.